Scripts/download-portal-item-data.swift

```diff
@@ -200,16 +237,38 @@ portalItems.forEach { (portalURLString, portalItems) in
     portalItems.forEach { (portalItem) in
         // Have we already downloaded the item?
         let filename = downloadedItems[portalItem.identifier] ?? portalItem.filename
-        if FileManager.default.fileExists(atPath: destinationURLProvider.makeURL(filename: filename).path) {
+        
+        // Check if it is a non-archive single file.
+        let isFileExist: Bool = FileManager.default.fileExists(atPath: destinationURLProvider.makeURL(filename: filename).path)
+        
+        // Check if there is a sub-folder for the corresponding archive, and the sub-folder is empty or not.
+        // The corresponding sub-folder has the same name as the archive without the extension.
+        let subFolderURL = destinationURLProvider.makeSubFolderURLForArchive(folderName: (filename as NSString).deletingPathExtension)
+        let paths = try? FileManager.default.contentsOfDirectory(
+            at: subFolderURL,
+            includingPropertiesForKeys: nil,
+            options: .skipsHiddenFiles
+        )
+        /// true: the folder exists and is empty;
+        /// false: the folder exists and has content;
+        /// nil: the folder does not exist or not accessible
+        let isEmptySubFolder = paths?.isEmpty
+        
+        if isEmptySubFolder != nil && !isEmptySubFolder! {
+            // One exception to this case is that the content in the folder is changed. There is no way to detect that
+            // unless we keep track of each extracted file in the .downloaded_items dictionary.
+            print("Item \(portalItem.identifier) has already been downloaded, and is extracted to an folder")
+            downloadedItems[portalItem.identifier] = filename
```

Comment:
This seems like overkill. The archive name should be unique, so I don't think we need to check that the directory is empty. If it exists, we can assume it has been downloaded.